src/index.ts

```diff
@@ -184,7 +184,10 @@ export async function retry<T> (
           resolve(result);
         }).catch((err: any) => {
           clearTimeout(timer);
-          resolve(onError(err));
+          // Calling resolve with a Promise that rejects here will result
+          // in an unhandled rejection. Calling `reject` with errors
+          // does not result in an unhandled rejection
+          onError(err).then(resolve).catch(reject);
```

Comment:
is `onError` typed as `async`?

Reply:
Yes it is - https://github.com/lifeomic/attempt/pull/50/files/39f2d292cfd63e0ef814913594e47ac7f27c12b5#diff-a2a171449d862fe29692ce031981047d7ab755ae7f84c707aef80701b3ea0c80R140